File: Cat/src/main/java/com/carlos/cat/uiautomator/DebugBridge.java
/*
 * Copyright (C) 2012 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.carlos.cat.uiautomator;

import com.android.SdkConstants;
import com.android.ddmlib.AndroidDebugBridge;
import com.android.ddmlib.IDevice;
import com.carlos.cat.util.CConstants;

import java.io.File;

public class DebugBridge {
    private static AndroidDebugBridge sDebugBridge;
    public static String getAdbLocation() {

        // check if adb is present in platform-tools
        String toolDir = System.getenv(SdkConstants.ANDROID_HOME_ENV);
        String adbPath = toolDir + File.separator + SdkConstants.OS_SDK_PLATFORM_TOOLS_FOLDER + SdkConstants.FN_ADB;
        System.out.println("adbp:"+adbPath);
        if (new File(adbPath).exists()) {
            return adbPath;
        }

        // use config adb
        adbPath = CConstants.INSTANCE.getAdbPath();
        if (new File(adbPath).exists()) {
            return adbPath;
        }

        return "adb";
    }

    public static void init() {
        String adbLocation = getAdbLocation();
        System.out.println("adb:"+adbLocation);
        if (adbLocation != null) {
            AndroidDebugBridge.init(false /* debugger support */);
            sDebugBridge = AndroidDebugBridge.createBridge(adbLocation, false);
            //System.out.println(sDebugBridge);
        }
    }

    public static void terminate() {
        if (sDebugBridge != null) {
            sDebugBridge = null;
            AndroidDebugBridge.terminate();
        }
    }

    public static boolean isInitialized() {
        return sDebugBridge != null;
    }

    public static IDevice[] getDevices() {
        return sDebugBridge.getDevices();
    }
}

File: Cat/src/main/kotlin/com/carlos/cat/util/DebugBridge.kt
//package com.carlos.cat.util
//
//import com.android.SdkConstants
//import com.android.ddmlib.AndroidDebugBridge
//import com.android.ddmlib.IDevice
//import org.slf4j.LoggerFactory
//import java.io.File
//
///**
// * Created by Carlos on 2018/7/3.
// */
//object DebugBridge {
//
//    private val logger = LoggerFactory.getLogger(AdbHelper.javaClass)
//
//    private lateinit var androidDebugBridge: AndroidDebugBridge
//
//    fun init() {
//        val adbLocation = getAdbLocation()
//        logger.info("adbLocation:$adbLocation")
//        AndroidDebugBridge.init(false /* debugger support */)
//        androidDebugBridge = AndroidDebugBridge.createBridge(adbLocation, false)
//    }
//
//    fun getAdbLocation(): String {
//
//        // check if adb is present in platform-tools
//        val toolDir = System.getenv(SdkConstants.ANDROID_HOME_ENV)
//        var adbPath = toolDir + File.separator + SdkConstants.OS_SDK_PLATFORM_TOOLS_FOLDER + SdkConstants.FN_ADB
//        if (File(adbPath).exists()) {
//            return adbPath
//        }
//
//        // use config adb
//        adbPath = CConstants.adbPath
//        System.out.println("adb:"+adbPath)
//        if (File(adbPath).exists()) {
//            return adbPath
//        }
//
//        return "adb"
//    }
//
//    fun terminate() {
//        AndroidDebugBridge.terminate()
//    }
//
//    fun getDevices(): Array<IDevice> {
//        return androidDebugBridge.devices
//    }
//
//}